### PR TITLE
Add install instructions for ArchLinux

### LINKED ISSUE #2056: Add install instruction for ArchLinux?
I'm not an ArchLinux user and also don't know how AUR (ArchLinux User Repository) works.

There are two gmt packages on AUR:

- https://aur.archlinux.org/packages/gmt/
- https://aur.archlinux.org/packages/gmt6

Both are well-maintained and provide the latest 6.0.0 release. I think we can add it to INSTALL.md.

I know the second package is maintained by @holishing, but it seems the first package is more popular (have more votes).

### PR DESCRIPTION
**Pull request by holishing**
**Description of proposed changes**

Fixes #2056

**Reminders**

- [x] Correct base branch selected? **`master` for new features**
- [x] Make sure that your code follows our style. Use the other functions/files as a basis.

### CODE REVIEW

**Comment by seisman** [comment]
@holishing Thanks for your work. A few questions/comments:

1. Does AUR provide binary packages? Or Do users have to build gmt from source using the `makepkg` command?
2. Why is "python-sphinx" needed? The GMT source tarballs have the HTML documentations and manpages included (in the `doc_release` and `man_release` directories). Usually users don't need to build documentations themselves.

**Comment by holishing** [comment]
1. Packages in [AUR website](https://aur.archlinux.org/) does not provide binary packages officially.  Although there's third party binary AUR package builds, such as [ArchlinuxCN](https://www.archlinuxcn.org/archlinux-cn-repo-and-mirror/), but due to some build depedencies in PKGBUILD of AUR repo is lost, binary packages of gmt are still not available.

2. Thanks for your notice, I will try to package documentations file in the doc_release and man_release directories, and fix related package script.

**Comment by mcpca** [comment]
Hi,

Thanks for your work on these instructions and on the PKGBUILD.
Two suggestions:

- change the "upgrade system packages" command to `sudo pacman -Syu`, as that is the [recommended way](https://wiki.archlinux.org/index.php/Pacman#Upgrading_packages) to upgrade all official packages in an Arch Linux system;
- it could make sense to mention the gmt-{coast,cpt-city,dcw} AUR packages.

Cheers,
Miguel

**Comment by holishing** [comment]
Applied better `pacman` upgrade instruction, recommended related packages.
I'll correct `python-sphinx` related instructions after tested and build related AUR on Archlinux platforn,

**Comment by seisman** [comment]
@holishing Any updates on this?

**Comment by holishing** [comment]
Refined related instructions.

File: INSTALL.md [hunk 1]
```diff
@@ -224,6 +225,41 @@ Install other GMT dependencies (some are optional) via:
 include the supplements. If you need them you will need to
 [build from source](BUILDING.md) or upgrade to 18.04 LTS.
 
+## ArchLinux
```
Comment (seisman) [inline INSTALL.md]:
```suggestion
### ArchLinux
```

File: INSTALL.md [hunk 2]
```diff
@@ -224,6 +225,41 @@ Install other GMT dependencies (some are optional) via:
...
+
+### Install via building AUR package
```
Comment (seisman) [inline INSTALL.md]:
```suggestion
```

File: INSTALL.md [hunk 3]
```diff
@@ -224,6 +225,41 @@ Install other GMT dependencies (some are optional) via:
...
+    sudo pacman -Syu
+
+    # Install tools for building AUR packages
+    sudo pacman -S base-devel
+
+    # Use git command to clone Non-official AUR repo of gmt
+    # here's two options, **you just have to clone one of them**
+    # If you encounter some problems in those repo
+    # you can report to the authors first, or choose another one to clone
+
```
Comment (seisman) [inline INSTALL.md]:
```suggestion
```

**Comment by seisman** [comment]
@holishing Thanks. It looks good to me. Just a few tiny comments.

File: INSTALL.md [hunk 4]
```diff
@@ -224,6 +225,39 @@ Install other GMT dependencies (some are optional) via:
 include the supplements. If you need them you will need to
 [build from source](BUILDING.md) or upgrade to 18.04 LTS.
 
+### ArchLinux
+
+
```
Comment (seisman) [inline INSTALL.md]:
```suggestion
```

File: INSTALL.md [hunk 5]
```diff
@@ -224,6 +225,38 @@ Install other GMT dependencies (some are optional) via:
...
+It's easier to manage installed content and dependencies than directly
+building and manually `make install` in ArchLinux.
+
+    # Full update system packages first
+    sudo pacman -Syu
+
+    # Install tools for building AUR packages
+    sudo pacman -S base-devel
+
+    # Use git command to clone Non-official AUR repo of gmt
```
Comment (seisman) [inline INSTALL.md]:
```suggestion
    # Use git command to clone non-official AUR repo of gmt
```

File: INSTALL.md [hunk 6]
```diff
@@ -224,6 +225,38 @@ Install other GMT dependencies (some are optional) via:
...
+building and manually `make install` in ArchLinux.
+
+    # Full update system packages first
+    sudo pacman -Syu
+
+    # Install tools for building AUR packages
+    sudo pacman -S base-devel
+
+    # Use git command to clone non-official AUR repo of gmt
+    # here's two options, **you just have to clone one of them**
```
Comment (seisman) [inline INSTALL.md]:
```suggestion
    # Here are two options, **you just have to clone one of them**
```

File: INSTALL.md [hunk 7]
```diff
@@ -224,6 +225,38 @@ Install other GMT dependencies (some are optional) via:
...
+
+    # Full update system packages first
+    sudo pacman -Syu
+
+    # Install tools for building AUR packages
+    sudo pacman -S base-devel
+
+    # Use git command to clone non-official AUR repo of gmt
+    # Here are two options. **You just have to clone one of them:**
+    # If you encounter some problems in those repo
```
Comment (seisman) [inline INSTALL.md]:
```suggestion
    # If you encounter some problems in those repo,
```

**Comment by seisman** [comment]
@holishing I've made commits directly to this branch and merged this PR. Thanks.

**Pull request merged**